app/services/browser_support.rb

```diff
@@ -38,10 +37,20 @@ def clear_cache!
 
     attr_reader :cache
 
-    def matchers
+    def browserslist_to_browser_map(browser)
+      if browser.ios?
+        BROWSERSLIST_TO_BROWSER_MAP.slice(:ios_saf)
+      elsif browser.platform.android_webview?
+        BROWSERSLIST_TO_BROWSER_MAP.slice(:android)
+      else
+        BROWSERSLIST_TO_BROWSER_MAP
+      end
+    end
+
+    def matchers(mapping)
       @matchers ||= browser_support_config.flat_map do |config_entry|
         key, version = config_entry.split(' ', 2)
-        browser_matcher = BROWSERSLIST_TO_BROWSER_MAP[key]
+        browser_matcher = mapping[key.to_sym]
```

Comment:
if the passed in `mapping` is now variable, that will mess with the memoization here? if we memoize based on the result of a cached iOS or Android one, then that will affect future calls?

Reply:
Ooh, yeah, good call. At least at a glance I feel like that would be an issue.

Reply:
maybe we memoize the full set of mappings from the constant still and do `matchers.slice()`?

Reply:
Not sure it's exactly what you had in mind, but I think 7511389264 should address this. The main challenge is that the `matchers` shape as an array made it difficult to "pick" specific browser matchers, so I restructured it as a hash.